mir_eval/separation.py

```diff
@@ -114,6 +108,12 @@ def validate(reference_sources, estimated_sources):
                                                 MAX_SOURCES))
 
 
+def _any_source_silent(sources):
+    """Returns true if the parameter sources has any silent first dimensions"""
+    return np.any(np.all(np.sum(
+        sources, axis=np.arange(2, sources.ndim)) == 0, axis=1))
```

Comment:
np.arange(2,2) returns [], but the axis arg to np.sum requires an int or a
tuple (not a list).
axis=tuple(np.arange(...)) will work, I think.

  DAn.

On Mon, Aug 15, 2016 at 2:11 AM, Erik <notifications@github.com> wrote:

> In mir_eval/separation.py:
>
> > @@ -114,6 +108,12 @@ def validate(reference_sources, estimated_sources):
> >                                                  MAX_SOURCES))
> >
> >
> > +def _any_source_silent(sources):
> > +    """Returns true if the parameter sources has any silent first dimensions"""
> > +    return np.any(np.all(np.sum(
> > +        sources, axis=np.arange(2, sources.ndim)) == 0, axis=1))
>
> It must return something other than an empty list for the null case.. I'll
> change that back and make the other style correction.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/craffel/mir_eval/pull/216/files/881639e4d0e4f92e23ec7ae1f8c8371a6788ec69#r74723412>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAhs0Q23poH4OvrbMLGuAQkLZj4baH5pks5qgAMYgaJpZM4JfNPY>
> .
>